### ISSUE TITLE
This plugin does not trigger end event correctly

### ISSUE THREAD

**Issue by jiyinyiyong** [jorrit/gulp-requirejs#5]
A possible solution is to add end here at about line 40:
```js
    // try {
        optimize(opts, function(text) {
            _s.write(new File({
                path: _fName,
                contents: new Buffer(text)
            }));
            _s.end() // <--- end state
        });
    // } catch (err) {
    //     _s.emit('error', err);
    // }
```

**Comment by MadLittleMods** [comment]
It looks like pull request #2 has a nice solution to this problem.

You can install it and depend on it via npm:
`npm install git://github.com/bleadof/gulp-requirejs#end-stream-and-emit-error --save-dev`

**Comment by ChiperSoft** [comment]
For those wanting a resolution that doesn't involve using a fork, I managed to solve it by hijacking the output using through2 and forcing a stream end.

```js
gulp.task('requirejs', function () {
  return rjs({
      mainConfigFile: 'public/assets/require.config.js',
      baseUrl: 'public/assets/',
      preserveLicenseComments: false,
      name: '../main',
      out: 'main.js'
    })
    .pipe(through2.obj(function (file, enc, next) {
      this.push(file);
      this.end();
      next();
    }))
    .pipe(uglify())
    .pipe(gulp.dest('./public/assets/'));
});
```

**Comment by jiyinyiyong** [comment]
@RobinThrift a lot of people are using this plugin, can you fix this now?

```
341 downloads in the last day
1,967 downloads in the last week
5,801 downloads in the last month
```

**Comment by ebaranov** [comment]
@jiyinyiyong +1

**Comment by zensh** [comment]
Here is a perfect version:

```js
'use strict';

var through = require('through2');
var gutil = require('gulp-util');
var PluginError = gutil.PluginError;
var requirejs  = require('requirejs');

// consts
var PLUGIN_NAME = 'gulp-rjs';

// exporting the plugin main function
module.exports = rjs;

function rjs(options) {
  var stream = through.obj();
  var path = options.out;

  options.out = function(string) {
    stream.push(new gutil.File({
      path: path,
      contents: new Buffer(string)
    }));
  };

  requirejs.optimize(options, function(data) {
    stream.push(null);
  }, function(err) {
    stream.emit('error', err);
  });

  return stream;
}

```

**Comment by R4YM3** [comment]
+1

**Comment by jorrit** [comment]
Fixed in #20.
